src/layers/alibi_embedding.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <iostream>
+
+class AlibiEmbedding {
+public:
+    AlibiEmbedding(const int headNum, const int seqLen);
+
+    ~AlibiEmbedding() {
+        maxLen = 0;
+        maxHeadNums = 0;
+        free(posMatrix);
+        free(slopeM);
+    }
+
+    void alibiGetRelativePos(const int seqLen);
+
+    void alibiGetSlope(const int headNum);
+
+    // headIdx is [0,n]
+    void alibiGetBias(const int headIdx, const int seqLen, float *bias_matrx);
+
+private:
+    static bool initialized;
```

Comment:
What's the meaning of this bool var? Nothing will be shared between objects.

Reply:
This variable is used to determine whether the distance matrix has been initialized or not. And In this pr, 1. If alibiGetBias function is called directly without the initialization judgment, an error will occur. 2. The distance matrix can be reused w/ single instance.

Reply:
All the functions are not static, so they can't be called without instantiating a class. And you will initialize it in the constructor. Will you offer a constructor method without initialization or a static method?